realm/realm-library/proguard-rules-consumer-common.pro

```diff
@@ -22,5 +22,9 @@
 -dontnote rx.Observable
 
 # Referenced from JNI
--keep class org.bson.types.Decimal128
--keep class org.bson.types.ObjectId
\ No newline at end of file
+-keep class org.bson.types.Decimal128 {
```

Comment:
This also keeps the class name itself, right?

Reply:
Yes, it initially kept just the class name. With this PR we protect the constructors called from JNI.